lib/elaborate/elaborate-selected-names.ts

```diff
@@ -0,0 +1,73 @@
+import { DiagnosticSeverity } from "vscode-languageserver";
+import { implementsIHasReference, implementsIHasTypeReference } from "../parser/interfaces";
+import { OArray, ObjectBase, OFile, ORecord, ORecordChild, OSelectedName, OSelectedNameRead, OSelectedNameWrite } from "../parser/objects";
+
+function findRecordElement(file: OFile, selectedName: OSelectedName | OSelectedNameWrite | OSelectedNameRead, typeDefinition: ObjectBase) {
+  if (typeDefinition instanceof ORecord) {
+    let found = false;
+    for (const child of typeDefinition.children) {
+      if (child.lexerToken.getLText() === selectedName.referenceToken.getLText()) {
+        selectedName.definitions.push(child);
+        found = true;
+      }
+    }
+    if (found === false) {
+      file.parserMessages.push({
+        message: `${selectedName.referenceToken} does not exist on record ${typeDefinition.lexerToken}`,
+        range: selectedName.referenceToken.range,
+        severity: DiagnosticSeverity.Error
+      });
+    }
```

Comment:
Why is this a parserMessage?
Should something like this: vhdlLinter.addMessage({
                  range: useClause.range,
                  severity: DiagnosticSeverity.Warning,
                  message: `could not find instantiated package from package ${useClause.library}.${useClause.packageName}`
                }, 'elaborate');

Reply:
Ah, yes makes more sense. will change